src/lsm/segmented_array.zig

```diff
@@ -1253,7 +1253,11 @@ fn FuzzContextType(
                     context.array.search(math.maxInt(Key)),
                     .ascending,
                 );
-                try testing.expectEqual(iterator_end.next(), null);
+                try testing.expectEqual(
```

Comment:
I wanted to add symmetric case for 0, but it looks like that actually reveals a bug in our code :fearful: 

https://github.com/tigerbeetle/tigerbeetle/pull/1887

Reply:
Ah, no, this is surprising, but I think correct. The second commit adds this to a fuzzer in the right way.